File: java/src/main/java/ps_traning/programmers/level_3/네트워크.java
package ps_traning.programmers.level_3;

import java.util.Stack;

public class 네트워크 {
    public int solution(int n, int[][] computers) {
        int answer = 0;
        boolean[] visited = new boolean[n];
        for (int i = 0; i < n; i++) {
            if (visited[i]) continue;
            answer++;
            Stack<Integer> st = new Stack<>();
            st.push(i);
            visited[i] = true;
            while (!st.isEmpty()) {
                int current = st.pop();
                for (int j = 0; j < n; j++) {
                    if (computers[current][j] == 1 && !visited[j]) {
                        visited[j] = true;
                        st.push(j);
                    }
                }
            }
        }
        return answer;
    }
}

File: python/with_java/programmers/level 3/네트워크.py
def solution(n, computers):
    answer = 0
    visited = [False for _ in range(n)]
    for i in range(n):
        if visited[i]:
            continue
        answer += 1
        stack = [i]
        while stack:
            current = stack.pop()
            visited[current] = True
            for j in range(n):
                if computers[current][j] == 1 and not visited[j]:
                    stack.append(j)
                    visited[j] = True
    return answer